engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1509,41 +1518,51 @@ private static boolean serializeAllTable(
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        jobScheduler.iterateParallel(
+                executionContext,
+                logOutput -> logOutput.append("serializeAllTable"),
+                JobScheduler.DEFAULT_CONTEXT_FACTORY,
+                0, columnSources.length,
+                (context, colIdx, nestedErrorConsumer) -> serializeColumn(columnSources, colIdx, usePrev, snapshot,
+                        table, logIdentityObject, columnsToSerialize),
+                () -> waitForResult.complete(Boolean.TRUE),
+                err -> {
+                    exception.set(err);
+                    waitForResult.complete(Boolean.FALSE);
+                });
+        try {
+            final boolean ret = waitForResult.get();
+            if (!ret) {
+                final Exception err = exception.get();
+                if (!(err instanceof SnapshotUnsuccessfulException)) {
+                    throw new SnapshotUnsuccessfulException("Snapshot failed", err);
                 }
-
-                final ColumnSource<?> columnSource = table.getColumnSource(columnSources[ii]);
-
-                final BarrageMessage.AddColumnData acd = new BarrageMessage.AddColumnData();
-                snapshot.addColumnData[ii] = acd;
-                final boolean columnIsEmpty = columnsToSerialize != null && !columnsToSerialize.get(ii);
-                final RowSet rows = columnIsEmpty ? RowSetFactory.empty() : snapshot.rowsIncluded;
-                // Note: cannot use shared context across several calls of differing lengths and no sharing necessary
-                // when empty
-                final ColumnSource<?> sourceToUse = ReinterpretUtils.maybeConvertToPrimitive(columnSource);
-                acd.data = getSnapshotDataAsChunkList(sourceToUse, columnIsEmpty ? null : sharedContext, rows, usePrev);
-                acd.type = columnSource.getType();
-                acd.componentType = columnSource.getComponentType();
-                acd.chunkType = sourceToUse.getChunkType();
-
-                final BarrageMessage.ModColumnData mcd = new BarrageMessage.ModColumnData();
-                snapshot.modColumnData[ii] = mcd;
-                mcd.rowsModified = RowSetFactory.empty();
-                mcd.data = getSnapshotDataAsChunkList(sourceToUse, null, RowSetFactory.empty(), usePrev);
-                mcd.type = acd.type;
-                mcd.componentType = acd.componentType;
-                mcd.chunkType = sourceToUse.getChunkType();
+                // Free any resources that may have been allocated
+                snapshot.close();
+                return false;
             }
+        } catch (final InterruptedException e) {
+            throw new java.util.concurrent.CancellationException("Interrupted while serializing table");
+        } catch (final ExecutionException e) {
+            throw new SnapshotUnsuccessfulException("Execution exception occurred while serializing table",
```

Comment:
We could skip using the `AtomicReference` for exception passing. The `ExecutionException`'s cause is an equivalent vector.
Here you could check for `SnapshotInconsistentException` and return `false. Otherwise, wrap in a sub-class of `RuntimeException`, like you're currently doing. Maybe a `SnapshotAttemptUnsuccessfulException` or `ColumnSnapshotUnsuccessfulException`.